lib/screens/UsersProfileScreen.dart

```diff
@@ -1349,93 +1275,54 @@ class _EditPasswordPageState extends State<EditPasswordPage> {
-                                            print("Incorrect formatting!");
-                                            break;
-                                          case 401:
-                                            errorMessage =
-                                            'Access token missing';
-                                            break;
-                                          case 403:
-                                            final changeToken = await Authentication.refreshToken();
-                                            switch (changeToken.statusCode) {
-                                              case 200:
-                                                var tokens = json.decode(
-                                                    changeToken.body);
-                                                user.accessToken =
-                                                tokens['accessToken'];
-                                                user.refreshToken =
-                                                tokens['refreshToken'];
-                                                break;
-                                              case 400:
-                                                errorMessage =
-                                                'Could not refresh tokens';
-                                                break;
-                                              case 401:
-                                                errorMessage = 'Token missing';
-                                                break;
-                                              case 404:
-                                                errorMessage = 'User not found';
-                                                break;
-                                            }
-                                            break;
-                                          case 404:
-                                            errorMessage = 'User not found';
-                                            break;
-                                          default:
-                                            print(
-                                                'Something in edit profile went wrong!');
-                                            break;
-                                        }
+                                  Map<String, dynamic> changes = {
+                                    'firstName': user.firstName.trim(),
+                                    'lastName': user.lastName.trim(),
+                                    'username': user.email.trim(),
+                                    'password': _newPassword.value.text.trim(),
+                                  };
+
+                                  try {
+                                    final res =
+                                        await User.updateUser(changes);
+                                    if (res.statusCode == 200) {
```

Comment:
I might be wrong, but I think 1290-1311 can be simplified - if this doesn't work, np, but you should be able to take advantage of `Navigator.pop(context)`
```dart
if (res.statusCode == 200) {
  user.password = _newPassword.value.text.trim();
  errorMessage == 'Successfully updated your password!';
  await Future.delayed(Duration(seconds: 1));
  clearFields();
  Navigator.pop(context);
}
// Note (don't need to include this in code): we only get to this code if the above
// is not true, since we leave the page after successful execution of the above
errorDialog(context);
```